meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot/0002-iMX8M-default-make-builds-SPL.patch

```diff
@@ -0,0 +1,143 @@
+From 748ad9e0f91b95b8e6065b9666e7f6ed997c1eb3 Mon Sep 17 00:00:00 2001
+From: Jorge Ramirez-Ortiz <jorge@foundries.io>
+Date: Tue, 26 Jan 2021 22:44:05 +0200
+Subject: [PATCH 2/2] iMX8M: default make builds SPL
+
+The flash.bin image therefore becomes the SPL image ready to be
+signed.
+
+Upstream-Status: Inappropriate [lmp specific]
+Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>
+---
+ iMX8M/soc.mak          | 33 ++++++++++++++++++++++++++
+ iMX8M/u-boot.its.patch | 53 ++++++++++++++++++++++++++++++++++++++++++
+ 2 files changed, 86 insertions(+)
+ create mode 100644 iMX8M/u-boot.its.patch
+
+diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
+index b7b3986..eb143f8 100644
+--- a/iMX8M/soc.mak
++++ b/iMX8M/soc.mak
+@@ -88,6 +88,20 @@ lpddr4_dmem_1d = lpddr4_pmu_train_1d_dmem$(LPDDR_FW_VERSION).bin
+ lpddr4_imem_2d = lpddr4_pmu_train_2d_imem$(LPDDR_FW_VERSION).bin
+ lpddr4_dmem_2d = lpddr4_pmu_train_2d_dmem$(LPDDR_FW_VERSION).bin
+ 
++################################
++# spl.bin + dtb + pad + ddr_fw #
++################################
++u-boot-signed-spl-ddr.bin: u-boot-spl.bin $(lpddr4_imem_1d) $(lpddr4_dmem_1d) $(lpddr4_imem_2d) $(lpddr4_dmem_2d)
```

Comment:
I'm going to explore adding the right dependencies to imx-boot recipe so that the build order is like this:
- atf compile / deploy
- op-tee compile / deploy
- u-boot compile: generates SPL and u-boot.bin
- use uboot-fitimage class to create u-boot.itb (signing the u-boot-spl.dtb in the process)
- build imx-boot for SPL only using deployed SPL.bin (signed dtb)

If we can do that then we can drop all of the u-boot.itb handling / u-boot.its patches here.  This ensures as we make changes to the uboot-fitimage class in the future, they are still being used by imx8*.